### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by SerhiiShapovalenko**

### CODE REVIEW

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
```
Comment (OleziO) [inline src/transformState.js]: Move all cases to the object.

**Review by OleziO — CHANGES_REQUESTED** [review]

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,37 @@
...
 function transformState(state, actions) {
-  // write code here
+  const actionsType = {
+    add: 'addProperties',
+    remove: 'removeProperties',
+    clear: 'clear',
+  };
+
+  for (const action of actions) {
+    switch (action.type) {
```
Comment (anstsot) [inline src/transformState.js:15]: don't forget about default case for `switch/case`

**Review by anstsot — CHANGES_REQUESTED** [review]
It’s almost done! Let’s improve a little

File: src/transformState.js [hunk 3]
```diff
@@ -5,7 +5,40 @@
...
+        for (const key of action.keysToRemove) {
+          if (state.hasOwnProperty(key)) {
+            delete state[key];
+          }
+        }
+        break;
+
+      case actionsType.clear:
+        for (const key in state) {
+          if (state.hasOwnProperty(key)) {
```
Comment (loralevitska) [inline src/transformState.js:30]: Of course, this key is in the object, but you get it from the object through a loop. Remove redundant checker

File: src/transformState.js [hunk 4]
```diff
@@ -5,7 +5,40 @@
...
+
+  for (const action of actions) {
+    switch (action.type) {
+      case actionsType.add :
+        Object.assign(state, action.extraData);
+        break;
+
+      case actionsType.remove:
+        for (const key of action.keysToRemove) {
+          if (state.hasOwnProperty(key)) {
```
Comment (loralevitska) [inline src/transformState.js:22]: Same here

**Review by loralevitska — APPROVED** [review]

**Pull request closed without merging**